eclass/toolchain.eclass

```diff
@@ -576,6 +576,17 @@ toolchain_pkg_pretend() {
 		_tc_use_if_iuse objc++ && \
 			ewarn 'Obj-C++ requires a C++ compiler, disabled due to USE="-cxx"'
 	fi
+
+	if in_iuse time64 && ! use time64 &&
+		has_version -r "${CATEGORY}/${PN}[time64(-)]" &&
+		[[ ! ${TC_FORCE_TIME32} ]]
```

Comment:
Are you sure this will work right for binpkgs? Check out net-misc/openssh and the trouble we had there.

Reply:
No clue, didn't think that far. Does Portage really wipe all envvars that were unset at build time? I thought it would only happen if the variable was set and you want to change the value.

I'm going to experiment with a dummy package.

Reply:
Sigh, indeed it wipes external envvars. So misdesigned…